detect_secrets/core/scan.py

```diff
@@ -327,25 +340,33 @@ def _process_line_based_plugins(
         ):
             continue
 
-        yield from (
-            secret
-            for plugin in get_plugins()
+        for plugin in get_plugins():
             for secret in _scan_line(
-                plugin=plugin,
-                filename=filename,
-                line=line,
-                line_number=line_number,
-                context=code_snippet,
-            )
-            if not _is_filtered_out(
-                required_filter_parameters=['context'],
-                filename=secret.filename,
-                secret=secret.secret_value,
-                plugin=plugin,
-                line=line,
-                context=code_snippet,
-            )
-        )
+                    plugin=plugin,
+                    filename=filename,
+                    line=line,
+                    line_number=line_number,
+                    context=code_snippet,
+            ):
+                if not _is_filtered_out(
+                        required_filter_parameters=['context'],
+                        filename=secret.filename,
+                        secret=secret.secret_value,
+                        plugin=plugin,
+                        line=line,
+                        context=code_snippet,
+                ):
+                    if determine_file_type(filename) == FileType.YAML and secret.secret_value:
```

Comment:
Have you looked at [yaml transformer](https://github.com/Yelp/detect-secrets/blob/master/detect_secrets/transformers/yaml.py)? Yaml files are transformed according to this transformer. So any changes should be done here.

Reply:
Hi @jpdakran !
Yes I have, it seems it merges the lines into 1 which makes sense from scanning perspective, but not from finding perspective. Otherwise, the secret's line is the first one, even though the secret is on the third line.
The yaml transformer doesn't retain a mapping of this...